Fix package metadata

jakirkham opened the pull request:
Corrects some copy-paste errors in the package metadata.

<!--
Thank you for pull request.
Below are a few things we ask you kindly to self-check before getting a review. Remove checks that are not relevant.
-->
Checklist
* [ ] Used a fork of the feedstock to propose changes
* [ ] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [ ] [Re-rendered]( https://conda-forge.org/docs/conda_smithy.html#how-to-re-render ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [ ] Ensured the license file is being packaged.

<!--
Please note any issues this fixes using [closing keywords]( https://help.github.com/articles/closing-issues-using-keywords/ ):
-->

<!--
Please add any other relevant info below:
-->

jakirkham committed:
Update summary and drop description               [ci skip]

jakirkham committed:
Drop doc URL (as it is inaccurate)                [ci skip]

conda-forge-linter commented:
Hi! This is the friendly automated conda-forge-linting service.

I just wanted to let you know that I linted all conda-recipes in your PR (```recipe```) and found it was in an excellent condition.

jakirkham committed:
MNT: Re-rendered with conda-build 3.17.5, conda-smithy 3.2.14, and conda-forge-pinning 2019.02.24  [ci skip]

Now that the metadata has been updated. Re-render to update the
feedstock README accordingly.
